### DISCUSSION TITLE
To set expireIn option makes relative time from local machine time.

### DISCUSSION THREAD

**Discussion by crabart** [deta/cloud-docs#486] [Ideas]
I think that when use expireIn option, to set __expires time is relative time from server time better than it from local machine time.

When using Deta Base, I thought that if I set expireIn option, the expiring time is set to exact time I set.
But, it expired early.

I tried to set __expires myself and to set using expireIn option.
__expires field was same time.
Here's the code(JavaScript) I tried.
```
db.put(
  {
    __expires: Date.now() / 1000 + 60,
  },
  "data1"
);

db.put(
  {
    data: "data",
  },
  "data2",
  { expireIn: 60 }
);
```
And I used deta version is 
```
"dependencies": {
  "deta": "^1.1.0"
}
```

**Comment by jnsougata** [comment]
Deta uses UNIX time stamp to generalize different time zones and handle the expiration ... It's neither server time nor local machine time afaik

**Reply by crabart** [reply]
Thank you for your reply.
I know Deta uses UNIX timestamp. And sorry that local machine means the computer which executes node(call it exec computer).
I understand __expires is generated by the exec computer.
[Here](https://github.com/deta/deta-javascript/blob/e64f24a1c54eee7c71a0098f876bb1387b0f807e/src/base/base.ts#L70)'s the code where to call method for generate __expires in put method.
And finally __expires generated by [here](https://github.com/deta/deta-javascript/blob/e64f24a1c54eee7c71a0098f876bb1387b0f807e/src/utils/date.ts#L21).
It uses Date of JavaScript with no arguments to make instance. And it initialized by the exec computer's time.
Extreme story, what I want to say, if the exec computer is set time before 1 day of server.
And I set expireIn: 86400(it is seconds of 1 day), but it expire soon.
